distributed/scheduler.py

```diff
@@ -3965,9 +3965,7 @@ def transition(self, key, finish, *args, **kwargs):
                     self.tasks[ts.key] = ts
                 for plugin in list(self.plugins):
                     try:
-                        if self.tasks[key].processing_on is not None:
-                            kwargs['worker'] = self.tasks[key].processing_on.address
-                        plugin.transition(key, start, finish2, *args, **kwargs)
+                        plugin.transition(key, start, finish2, task_state=ts, *args, **kwargs)
```

Comment:
I suspect that Python will yell at you here for placing `*args` after a named keyword argument.  You'll probably have to insert this into kwargs yourself.

Reply:
Good point, wasn't giving errors but looking at it now args was always empty so probably why. Sorry, github didn't auto refresh the second post.